RiotSwiftUI/Modules/LocationSharing/LiveLocationSharingViewer/View/LiveLocationSharingViewer.swift

```diff
@@ -34,23 +34,42 @@ struct LiveLocationSharingViewer: View {
+                               annotations: viewModel.viewState.annotations,
+                               highlightedAnnotation: viewModel.viewState.highlightedAnnotation,
+                               userAvatarData: nil,
+                               showsUserLocation: viewModel.viewState.showsUserLocation,
+                               userAnnotationCanShowCallout: true,
+                               userLocation: Binding.constant(nil),
+                               mapCenterCoordinate: Binding.constant(nil),
+                               onCalloutTap: { annotation in
+                                   if let userLocationAnnotation = annotation as? UserLocationAnnotation {
+                                       viewModel.send(viewAction: .share(userLocationAnnotation))
+                                   }
+                               },
+                               errorSubject: viewModel.viewState.errorSubject)
+    }
+    
     var body: some View {
         ZStack(alignment: .bottom) {
             if !viewModel.viewState.showMapLoadingError {
-                LocationSharingMapView(tileServerMapURL: viewModel.viewState.mapStyleURL,
-                                       annotations: viewModel.viewState.annotations,
-                                       highlightedAnnotation: viewModel.viewState.highlightedAnnotation,
-                                       userAvatarData: nil,
-                                       showsUserLocation: false,
-                                       userAnnotationCanShowCallout: true,
-                                       userLocation: Binding.constant(nil),
-                                       mapCenterCoordinate: Binding.constant(nil),
-                                       onCalloutTap: { annotation in
-                                           if let userLocationAnnotation = annotation as? UserLocationAnnotation {
-                                               viewModel.send(viewAction: .share(userLocationAnnotation))
-                                           }
-                                       },
-                                       errorSubject: viewModel.viewState.errorSubject)
+                ZStack(alignment: .topTrailing) {
+                    mapView
+                    if !viewModel.viewState.isCurrentUserShared {
+                        Button {
+                            viewModel.send(viewAction: .showUserLocation)
+                        } label: {
+                            Image(uiImage: Asset.Images.locationCenterMapIcon.image)
+                                .foregroundColor(theme.colors.accent)
+                        }
+                        .padding(8.0)
+                        .background(theme.colors.background)
+                        .clipShape(Circle())
+                        .shadow(radius: 2.0)
+                        .offset(x: -11.0, y: 52)
+                    }
```

Comment:
it would nice to create another computed property for the button like you did for the `mapView`.
After that the `ZStack` should look much more readable.

Reply:
Would be quite nice to use `.overlay` in that case too. `ZStack`s with 2 items always seem weird to me.

Reply:
The only problem I see is that `.overlay` is iOS 15+ :-/

Reply:
Yeah this is the weird thing where Xcode hides the deprecated ones. The old one works on 13+:
https://developer.apple.com/documentation/swiftui/view/overlay(_:alignment:)

Reply:
done with old .overlay, it's much more clearer now